spec/classes/ntp_spec.rb

```diff
@@ -3,39 +3,30 @@
 describe 'ntp' do
   let(:facts) {{ :is_virtual => 'false' }}
 
-  ['Debian', 'RedHat', 'Fedora', 'Suse11', 'Suse12', 'FreeBSD', 'Archlinux', 'Gentoo', 'Gentoo (Facter < 1.7)'].each do |system|
-    context "when on system #{system}" do
-      let :facts do
-        case system
-        when 'Gentoo (Facter < 1.7)'
-          super().merge({ :osfamily => 'Linux', :operatingsystem => 'Gentoo' })
-        when 'Suse11'
-          super().merge({ :osfamily => 'Suse', :operatingsystem => 'SLES', :operatingsystemmajrelease => '11' })
-        when 'Suse12'
-          super().merge({ :osfamily => 'Suse', :operatingsystem => 'SLES', :operatingsystemmajrelease => '12' })
-        when 'Fedora'
-          super().merge({ :osfamily => 'RedHat', :operatingsystem => system, :operatingsystemmajrelease => '22' })
-        else
-          super().merge({ :osfamily => system, :operatingsystem => system})
-        end
+  on_supported_os.select { |_, f| f[:os]['family'] != 'Solaris' }.each do |os, f|
+    context "on #{os}" do
+      let(:facts) do
+        f.merge(super())
```

Comment:
You've reversed the merge operation here, does that still produce the desired output? In the previous implementation, it merged the os specific facts into `super()` and now it is merging `super()` into the os facts. This merge will overwrite any duplicate keys in `f` with the value in `super()` with this implementation... even `nil` values.